modules/perception/inference/libtorch/torch_net.cc

```diff
@@ -0,0 +1,108 @@
+/******************************************************************************
+ * Copyright 2020 The Apollo Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+
+#include "modules/perception/inference/libtorch/torch_net.h"
+
+#include "cyber/common/log.h"
+
+namespace apollo {
+namespace perception {
+namespace inference {
+
+TorchNet::TorchNet(const std::string &net_file, const std::string &model_file,
+                   const std::vector<std::string> &outputs)
+    : net_file_(net_file), model_file_(model_file), output_names_(outputs) {}
+
+bool TorchNet::Init(const std::map<std::string, std::vector<int>> &shapes) {
+  if (gpu_id_ >= 0) {
+    device_type_ = torch::kCUDA;
+    device_id_ = gpu_id_;
+  } else {
+    device_type_ = torch::kCPU;
+  }
+
+  // Init net
+  torch::Device device(device_type_, device_id_);
+  net_ = torch::jit::load(model_file_, device);
+
+  for (auto name : output_names_) {
+    auto blob = std::make_shared<apollo::perception::base::Blob<float>>
+                                                          (1, 4, 1, 1);
+    blobs_.emplace(std::make_pair(name, blob));
+  }
+
+  for (auto name : input_names_) {
+    auto iter = shapes.find(name);
+    if (iter != shapes.end()) {
+      auto blob = std::make_shared<apollo::perception::base::Blob<float>>
+                                                          (iter->second);
+      blobs_.emplace(std::make_pair(name, blob));
+    }
+  }
+  return true;
+}
+
+TorchNet::TorchNet(const std::string &net_file, const std::string &model_file,
+                   const std::vector<std::string> &outputs,
+                   const std::vector<std::string> &inputs)
+    : net_file_(net_file),
+      model_file_(model_file),
+      output_names_(outputs),
+      input_names_(inputs) {}
+
+std::shared_ptr<apollo::perception::base::Blob<float>> TorchNet::get_blob(
+    const std::string &name) {
+  auto iter = blobs_.find(name);
+  if (iter == blobs_.end()) {
+    return nullptr;
+  }
+  return iter->second;
+}
+
+bool TorchNet::reshape() {
+  return true;
```

Comment:
Just return true?

Reply:
Just in order to keep same interface with old caffe inference moduel

Reply:
Since it's not a virtual function of the parent class, it does not have to be implemented. Unless you might need it in the future.

Reply:
Detection component is under developing, if the detection module is not used I will remove it from this class.

Reply:
Got it.